app/src/lib.rs

```diff
@@ -49,22 +51,22 @@ fn switch<'a, G: Html>(ctx: ScopeRef<'a>, route: &'a ReadSignal<AppRoutes>) -> V
 /// # Props
 /// * `pathname` - Set to `Some(_)` if running on the server.
 #[component]
-pub fn App<G: Html>(ctx: ScopeRef, pathname: Option<String>) -> View<G> {
+pub fn App<G: Html>(ctx: Scope, pathname: Option<String>) -> View<G> {
     match pathname {
         Some(pathname) => {
-            let route = AppRoutes::match_path(&pathname);
+            let route = AppRoutes::default().match_path(&pathname);
```

Comment:
Is there a better way to do this? 

If the `match_path` signature was `fn match_path(path: &str) -> Self` instead of `fn match_path(&self, path: &str) -> Self` we could write:
```rust
let route = AppRoutes::match_path(&pathname);
```
Trait signature is `pub trait Route: Sized + Default` so in theory we don't need depend on having `&self` available.